### DISCUSSION TITLE
Rúnar Bjarnason - Unison

### DISCUSSION THREAD

**Discussion by hipstersmoothie** [devtools-fm/devtools.fm#124] [Announcements]
## Background

- [Rúnar Twitter](https://twitter.com/runarorama?ref_src=twsrc%5Egoogle%7Ctwcamp%5Eserp%7Ctwgr%5Eauthor)
- [Unisons Twitter](https://twitter.com/unisonweb)
- [Unison Language](https://www.unison-lang.org)
- [Unison Code](https://github.com/unisonweb/unison)
- [Unison Share](https://share.unison-lang.org)
- [Unison Cloud](https://www.unison.cloud)

## Themes

- Function Programming
- Unison

## Intro

**Andrew**: Hello, welcome to the devtools.fm podcast. This is a podcast about developer tools and the people who make them. I'm Andrew and this is my co-host Justin.

**Justin**: Hey everyone! Our guest today is Rúnar Bjarnason. Rúnar is a co-founder of Unison Computing and one of the creators of the Unison programming language which will be our topic today. Rúnar we're so excited to have you on. Unison has came up a lot in past episodes as the prime example of the future of programming. Before digging into that though, would you like to tell our users a little more about yourself?

### Language

- [x] Unison is a very unique programming language. Could you give our listeners an overview of what makes it different from other programming languages they may be familiar with?
	- Content Addressability
	- Code in a database
	- Abilities (algebraic effects)
	- Distributed Computing
	- Distributed Storage
- [x] What are the goals of the language?
- [x] How would someone write a unison program?
- [x] Unison introduces a lot of new concepts all at once. Do you think that'll be harmful to the adoption of the language?

#### Content Addressability

- [x] How unique are the hashes? What are the chances of a collision?
- [x] Where are the functions stored? Is the content space shared? If I write the same function as another person is it like I added nothing at all?
- [x] How does collaboration work? (version control)
- [x] How does documentation work?
- [x] How does this help with dependency conflicts?
- [x] How does this help with builds and type checking?
- [x] How does this help with refactoring?
- [x] Does unison cache outputs?
- [x] How does this make distributed programming easier?

### Platform

- [x] Can you share code?
- [x] How do you deploy a unison application?

### Company

- [x] Unison Computing is a public benefit corp. What does that mean and how does that shape how you do business?
- [x] What's your ultimate goal with Unison Computing?

### Future

- [x] What's next for unison?
- [x] What do you hope the future of code looks like?

**Comment by just-be-dev** [comment]
Tooltips:
- https://github.com/Doist/typist
- https://metaphor.systems/

**Comment by hipstersmoothie** [comment]
https://github.com/BuilderIO/figma-html
https://github.com/charmbracelet/vhs

**Comment by runarorama** [comment]
Tooltips:

* Copilot: https://github.com/features/copilot
* Exercism: https://exercism.org/
* OB-4 https://teenage.engineering/products/ob-4
